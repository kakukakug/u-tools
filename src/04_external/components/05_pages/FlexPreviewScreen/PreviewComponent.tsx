import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { Colors } from "../../../styles/Colors";
import { createStyle } from "../../../../01_entity/models/flex_style";

const styles = StyleSheet.create({
  parent: {
    backgroundColor: "#eee",
    padding: 10,
    borderWidth: 1,
    borderColor: Colors.icon,
    flex: 1,
  },
  children: {
    backgroundColor: Colors.white,
    borderWidth: 1,
    borderColor: Colors.icon,
    padding: 2,
  },
  previewText: {
    color: Colors.text,
  },
});

const PreviewChild = (props) => {
  const { no, flex, flexGrow, flexShrink, flexBasis, alignSelf, text } = props;
  return (
    <View
      style={[
        styles.children,
        createStyle({
          flex: flex,
        }),
        no === 2
          ? createStyle({
              flexGrow: flexGrow,
              flexShrink: flexShrink,
              flexBasis: flexBasis,
              alignSelf: alignSelf,
            })
          : {},
      ]}>
      <Text
        style={[
          styles.previewText,
          no === 2 ? { color: Colors.notification } : {},
        ]}>
        {text}-{no}
      </Text>
    </View>
  );
};

export const PreviewComponent = (props) => {
  const {
    flexDirection,
    justifyContent,
    alignItems,
    flexWrap,
    alignContent,
    flex,
    flexGrow,
    flexShrink,
    flexBasis,
    alignSelf,
    text,
  } = props;
  return (
    <View
      style={[
        styles.parent,
        createStyle({
          flexDirection: flexDirection,
          justifyContent: justifyContent,
          alignItems: alignItems,
          flexWrap: flexWrap,
          alignContent: alignContent,
        }),
      ]}>
      <PreviewChild
        no={1}
        flex={flex}
        flexGrow={flexGrow}
        flexShrink={flexShrink}
        flexBasis={flexBasis}
        alignSelf={alignSelf}
        text={text}
      />
      <PreviewChild
        no={2}
        flex={flex}
        flexGrow={flexGrow}
        flexShrink={flexShrink}
        flexBasis={flexBasis}
        alignSelf={alignSelf}
        text={text}
      />
      <PreviewChild
        no={3}
        flex={flex}
        flexGrow={flexGrow}
        flexShrink={flexShrink}
        flexBasis={flexBasis}
        alignSelf={alignSelf}
        text={text}
      />
      <PreviewChild
        no={4}
        flex={flex}
        flexGrow={flexGrow}
        flexShrink={flexShrink}
        flexBasis={flexBasis}
        alignSelf={alignSelf}
        text={text}
      />
    </View>
  );
};
