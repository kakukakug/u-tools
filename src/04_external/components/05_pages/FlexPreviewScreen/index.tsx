import React, { useState, useEffect, useCallback } from "react";
import { ScrollView, StyleSheet, Text, View, Picker } from "react-native";

import { Colors } from "../../../styles/Colors";
import { Console } from "../../01_atoms/Console";

export const FlexPreviewScreen = () => {
  const [flexDirection, setFlexDirection] = useState("none");
  const [justifyContent, setJustifyContent] = useState("none");
  const [alignItems, setAlignItems] = useState("none");
  const [flexWrap, setFlexWrap] = useState("none");
  const [flex, setFlex] = useState("none");
  const [alignContent, setAlignContent] = useState("none");
  const [alignSelf, setAlignSelf] = useState("none");
  const [text, setText] = useState("child");
  const [flexGrow, setFlexGrow] = useState("none");
  const [flexShrink, setFlexShrink] = useState("none");
  const [flexBasis, setFlexBasis] = useState("none");
  const [styleText, setStyleText] = useState("");

  const createStyle = useCallback((propObj) => {
    let style = {};
    for (var key in propObj) {
      if (propObj[key] !== "none") {
        if (["flex", "flexGrow", "flexShrink"].includes(key)) {
          style[key] = Number(propObj[key]);
        } else if (key === "flexBasis" && propObj[key] !== "auto") {
          style[key] = Number(propObj[key]);
        } else {
          style[key] = propObj[key];
        }
      }
    }
    return style;
  }, []);

  const PreviewChild = (data) => {
    return (
      <View
        style={[
          styles.children,
          createStyle({
            flex: flex,
          }),
          data.no === 2
            ? createStyle({
                flexGrow: flexGrow,
                flexShrink: flexShrink,
                flexBasis: flexBasis,
                alignSelf: alignSelf,
              })
            : {},
        ]}>
        <Text
          style={[styles.previewText, data.no === 2 ? { color: "#e77" } : {}]}>
          {text}-{data.no}
        </Text>
      </View>
    );
  };

  const PreviewContainer = () => {
    return (
      <View
        style={[
          styles.parent,
          createStyle({
            flexDirection: flexDirection,
            justifyContent: justifyContent,
            alignItems: alignItems,
            flexWrap: flexWrap,
            alignContent: alignContent,
          }),
        ]}>
        <PreviewChild no={1} />
        <PreviewChild no={2} />
        <PreviewChild no={3} />
        <PreviewChild no={4} />
      </View>
    );
  };

  const SettingComponent = (data) => {
    const { name, value, onValueChange, selectionArray } = data;
    return (
      <View style={styles.flexRow}>
        <Text style={styles.stylePropText}>{name}</Text>
        <Picker
          selectedValue={value}
          style={styles.picker}
          itemStyle={styles.pickerItem}
          onValueChange={(itemValue) => {
            onValueChange(itemValue);
          }}>
          {selectionArray.map((label, index) => {
            return <Picker.Item label={label} value={value} key={index} />;
          })}
        </Picker>
      </View>
    );
  };
  const CustomerContainer = () => {
    return (
      <View style={styles.customerContainer}>
        <View style={styles.parentCustomer}>
          <Text style={styles.h1Text}>Parent</Text>
          <SettingComponent
            name={"flexDirection"}
            value={flexDirection}
            onValueChange={setFlexDirection}
            selectionArray={[
              "none",
              "column",
              "row",
              "column-reverse",
              "row-reverse",
            ]}
          />
          <SettingComponent
            name={"justifyContent"}
            value={justifyContent}
            onValueChange={setJustifyContent}
            selectionArray={[
              "none",
              "flex-start",
              "center",
              "flex-end",
              "space-around",
              "space-between",
              "space-evenly",
            ]}
          />
          <SettingComponent
            name={"alignItems"}
            value={alignItems}
            onValueChange={setAlignItems}
            selectionArray={[
              "none",
              "flex-start",
              "center",
              "flex-end",
              "stretch",
            ]}
          />
          <SettingComponent
            name={"flexWrap"}
            value={flexWrap}
            onValueChange={setFlexWrap}
            selectionArray={["none", "nowrap", "wrap", "wrap-reverse"]}
          />
          <SettingComponent
            name={"alignContent"}
            value={alignContent}
            onValueChange={setAlignContent}
            selectionArray={[
              "none",
              "flex-start",
              "flex-end",
              "center",
              "stretch",
              "space-between",
              "space-around",
            ]}
          />
        </View>
        <View style={styles.childCustomer}>
          <Text style={styles.h1Text}>Children</Text>
          <SettingComponent
            name={"text"}
            value={text}
            onValueChange={setText}
            selectionArray={[
              "child",
              "long text. mega super long text contents",
            ]}
          />
          <SettingComponent
            name={"flex"}
            value={flex}
            onValueChange={setFlex}
            selectionArray={["none", "1", "2"]}
          />
          <Text style={styles.h2Text}>
            only <Text style={{ color: "#e77" }}>child-2</Text> style prop
          </Text>
          <SettingComponent
            name={"flexGrow"}
            value={flexGrow}
            onValueChange={setFlexGrow}
            selectionArray={["none", "0", "1", "2", "3"]}
          />
          <SettingComponent
            name={"flexShrink"}
            value={flexShrink}
            onValueChange={setFlexShrink}
            selectionArray={["none", "1", "2", "3"]}
          />
          <SettingComponent
            name={"flexBasis"}
            value={flexBasis}
            onValueChange={setFlexBasis}
            selectionArray={["none", "auto", "10", "20", "30"]}
          />
          <SettingComponent
            name={"alignSelf"}
            value={alignSelf}
            onValueChange={setAlignSelf}
            selectionArray={[
              "none",
              "auto",
              "flex-start",
              "flex-end",
              "center",
              "stretch",
              "baseline",
            ]}
          />
        </View>
      </View>
    );
  };

  const styleTextFormat = useCallback(
    (obj) => {
      let styleObj = createStyle(obj);
      let ret = "";
      for (var key in styleObj) {
        if (typeof styleObj[key] === "string") {
          ret += "    " + key + ': "' + styleObj[key] + '"' + ",\n";
        } else {
          ret += "    " + key + ": " + styleObj[key] + ",\n";
        }
      }
      return ret;
    },
    [createStyle]
  );

  useEffect(() => {
    const parent = styleTextFormat({
      flexDirection: flexDirection,
      justifyContent: justifyContent,
      alignItems: alignItems,
      flexWrap: flexWrap,
      alignContent: alignContent,
    });
    const children = styleTextFormat({
      flex: flex,
    });
    const child2 = styleTextFormat({
      flexGrow: flexGrow,
      flexShrink: flexShrink,
      flexBasis: flexBasis,
      alignSelf: alignSelf,
    });
    setStyleText(`  parent: {
    flex: 1,
${parent}  },
  chidren: {
${children}  },
  child-2: {
${child2}  },`);
  }, [
    flexDirection,
    justifyContent,
    alignItems,
    flexWrap,
    alignContent,
    flex,
    flexGrow,
    flexShrink,
    flexBasis,
    alignSelf,
    styleTextFormat,
  ]);

  return (
    <View style={styles.container}>
      <ScrollView
        style={styles.scrollContainer}
        contentContainerStyle={styles.contentContainer}>
        <View style={styles.title}>
          <Text style={styles.titleText}>
            React Native FlexBox Style Creator
          </Text>
        </View>
        <View style={styles.main}>
          <View style={styles.settingContainer}>
            <Text style={styles.subTitleText}>Property selecter</Text>
            <CustomerContainer />
          </View>
          <View style={styles.previewContainer}>
            <Text style={styles.subTitleText}>Preview</Text>
            <PreviewContainer />
          </View>
        </View>
      </ScrollView>
      <Console consoleText={styleText} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  scrollContainer: {
    flex: 1,
  },
  contentContainer: {
    flex: 1,
    alignItems: "center",
  },
  main: {
    flex: 1,
    flexDirection: "row",
  },
  settingContainer: {
    padding: 10,
    width: 400,
  },
  previewContainer: {
    padding: 10,
    width: 400,
  },
  customerContainer: {
    flex: 3,
    flexDirection: "row",
    alignItems: "flex-start",
  },

  parentCustomer: {
    flex: 1,
    backgroundColor: Colors.surface,
    paddingHorizontal: 4,
    height: 300,
  },
  childCustomer: {
    flex: 1,
    backgroundColor: Colors.white,
    paddingHorizontal: 4,
    borderWidth: 1,
    borderColor: Colors.icon,
    height: 300,
  },

  parent: {
    backgroundColor: Colors.surface,
    padding: 10,
    flex: 1,
  },
  children: {
    backgroundColor: Colors.white,
    borderWidth: 1,
    borderColor: Colors.icon,
    padding: 2,
  },
  previewText: {
    color: Colors.text,
  },
  flexRow: {
    flexDirection: "row",
    justifyContent: "flex-end",
    alignItems: "center",
    paddingHorizontal: 4,
  },

  subTitleText: {
    fontSize: 26,
    color: Colors.icon,
    textAlign: "center",
    paddingTop: 6,
  },
  h1Text: {
    fontSize: 20,
    color: Colors.text,
    textAlign: "center",
    paddingTop: 6,
  },
  h2Text: {
    fontSize: 16,
    color: Colors.text,
    textAlign: "center",
    paddingTop: 10,
  },
  stylePropText: {
    fontSize: 14,
    color: "rgba(96,100,109, 1)",
    textAlign: "center",
    paddingHorizontal: 6,
    paddingVertical: 4,
  },

  picker: {
    height: 20,
    width: 60,
    borderRadius: 4,
  },
  pickerItem: {},

  title: { marginVertical: 10 },
  titleText: { fontSize: 38, color: Colors.text },
});
