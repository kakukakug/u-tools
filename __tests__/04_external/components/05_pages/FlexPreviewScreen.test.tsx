import React from "react";
import { render } from "@testing-library/react-native";

import { PreviewComponent } from "../../../../src/04_external/components/05_pages/FlexPreviewScreen/PreviewComponent";
import { SettingComponent } from "../../../../src/04_external/components/05_pages/FlexPreviewScreen/SettingComponent";

describe("PreviewComponent", () => {
  it(`PreviewComponentを描写`, () => {
    const component = render(
      <PreviewComponent
        flexDirection={"test"}
        justifyContent={"test"}
        alignItems={"test"}
        flexWrap={"test"}
        alignContent={"test"}
        flex={"test"}
        flexGrow={"test"}
        flexShrink={"test"}
        flexBasis={"test"}
        alignSelf={"test"}
        text={"test"}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
describe("SettingComponent", () => {
  it(`SettingComponentを描写`, () => {
    const mockSetState = jest.fn;
    const component = render(
      <SettingComponent
        flexDirection={"test"}
        justifyContent={"test"}
        alignItems={"test"}
        flexWrap={"test"}
        alignContent={"test"}
        flex={"test"}
        flexGrow={"test"}
        flexShrink={"test"}
        flexBasis={"test"}
        alignSelf={"test"}
        text={"test"}
        setFlexDirection={mockSetState}
        setJustifyContent={mockSetState}
        setAlignItems={mockSetState}
        setFlexWrap={mockSetState}
        setAlignContent={mockSetState}
        setFlex={mockSetState}
        setFlexGrow={mockSetState}
        setFlexShrink={mockSetState}
        setFlexBasis={mockSetState}
        setAlignSelf={mockSetState}
        setText={mockSetState}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
