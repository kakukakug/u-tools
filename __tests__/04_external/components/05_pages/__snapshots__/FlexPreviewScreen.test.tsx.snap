// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewComponent PreviewComponentを描写 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ccc",
        "flex": 1,
        "padding": 10,
      },
      Object {
        "alignContent": "test",
        "alignItems": "test",
        "flexDirection": "test",
        "flexWrap": "test",
        "justifyContent": "test",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fffffe",
          "borderColor": "#666",
          "borderWidth": 1,
          "padding": 2,
        },
        Object {
          "flex": NaN,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {},
        ]
      }
    >
      test
      -
      1
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fffffe",
          "borderColor": "#666",
          "borderWidth": 1,
          "padding": 2,
        },
        Object {
          "flex": NaN,
        },
        Object {
          "alignSelf": "test",
          "flexBasis": NaN,
          "flexGrow": NaN,
          "flexShrink": NaN,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {
            "color": "#e77",
          },
        ]
      }
    >
      test
      -
      2
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fffffe",
          "borderColor": "#666",
          "borderWidth": 1,
          "padding": 2,
        },
        Object {
          "flex": NaN,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {},
        ]
      }
    >
      test
      -
      3
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fffffe",
          "borderColor": "#666",
          "borderWidth": 1,
          "padding": 2,
        },
        Object {
          "flex": NaN,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {},
        ]
      }
    >
      test
      -
      4
    </Text>
  </View>
</View>
`;

exports[`SettingComponent SettingComponentを描写 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 3,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ccc",
        "flex": 1,
        "height": 300,
        "paddingHorizontal": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 20,
          "paddingTop": 6,
          "textAlign": "center",
        }
      }
    >
      Parent
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        flexDirection
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "column",
                "textColor": undefined,
                "value": "column",
              },
              Object {
                "label": "row",
                "textColor": undefined,
                "value": "row",
              },
              Object {
                "label": "column-reverse",
                "textColor": undefined,
                "value": "column-reverse",
              },
              Object {
                "label": "row-reverse",
                "textColor": undefined,
                "value": "row-reverse",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        justifyContent
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "flex-start",
                "textColor": undefined,
                "value": "flex-start",
              },
              Object {
                "label": "center",
                "textColor": undefined,
                "value": "center",
              },
              Object {
                "label": "flex-end",
                "textColor": undefined,
                "value": "flex-end",
              },
              Object {
                "label": "space-around",
                "textColor": undefined,
                "value": "space-around",
              },
              Object {
                "label": "space-between",
                "textColor": undefined,
                "value": "space-between",
              },
              Object {
                "label": "space-evenly",
                "textColor": undefined,
                "value": "space-evenly",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        alignItems
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "flex-start",
                "textColor": undefined,
                "value": "flex-start",
              },
              Object {
                "label": "center",
                "textColor": undefined,
                "value": "center",
              },
              Object {
                "label": "flex-end",
                "textColor": undefined,
                "value": "flex-end",
              },
              Object {
                "label": "stretch",
                "textColor": undefined,
                "value": "stretch",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        flexWrap
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "nowrap",
                "textColor": undefined,
                "value": "nowrap",
              },
              Object {
                "label": "wrap",
                "textColor": undefined,
                "value": "wrap",
              },
              Object {
                "label": "wrap-reverse",
                "textColor": undefined,
                "value": "wrap-reverse",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        alignContent
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "flex-start",
                "textColor": undefined,
                "value": "flex-start",
              },
              Object {
                "label": "flex-end",
                "textColor": undefined,
                "value": "flex-end",
              },
              Object {
                "label": "center",
                "textColor": undefined,
                "value": "center",
              },
              Object {
                "label": "stretch",
                "textColor": undefined,
                "value": "stretch",
              },
              Object {
                "label": "space-between",
                "textColor": undefined,
                "value": "space-between",
              },
              Object {
                "label": "space-around",
                "textColor": undefined,
                "value": "space-around",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fffffe",
        "borderColor": "#666",
        "borderWidth": 1,
        "flex": 1,
        "height": 300,
        "paddingHorizontal": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 20,
          "paddingTop": 6,
          "textAlign": "center",
        }
      }
    >
      Children
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        text
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "child",
                "textColor": undefined,
                "value": "child",
              },
              Object {
                "label": "long text. mega super long text contents",
                "textColor": undefined,
                "value": "long text. mega super long text contents",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        flex
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "1",
                "textColor": undefined,
                "value": "1",
              },
              Object {
                "label": "2",
                "textColor": undefined,
                "value": "2",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 16,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      only 
      <Text
        style={
          Object {
            "color": "#e77",
          }
        }
      >
        child-2
      </Text>
       style prop
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        flexGrow
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "0",
                "textColor": undefined,
                "value": "0",
              },
              Object {
                "label": "1",
                "textColor": undefined,
                "value": "1",
              },
              Object {
                "label": "2",
                "textColor": undefined,
                "value": "2",
              },
              Object {
                "label": "3",
                "textColor": undefined,
                "value": "3",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        flexShrink
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "1",
                "textColor": undefined,
                "value": "1",
              },
              Object {
                "label": "2",
                "textColor": undefined,
                "value": "2",
              },
              Object {
                "label": "3",
                "textColor": undefined,
                "value": "3",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        flexBasis
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "auto",
                "textColor": undefined,
                "value": "auto",
              },
              Object {
                "label": "10",
                "textColor": undefined,
                "value": "10",
              },
              Object {
                "label": "20",
                "textColor": undefined,
                "value": "20",
              },
              Object {
                "label": "30",
                "textColor": undefined,
                "value": "30",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 14,
            "paddingHorizontal": 6,
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        alignSelf
      </Text>
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 20,
            "width": 60,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "none",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "auto",
                "textColor": undefined,
                "value": "auto",
              },
              Object {
                "label": "flex-start",
                "textColor": undefined,
                "value": "flex-start",
              },
              Object {
                "label": "flex-end",
                "textColor": undefined,
                "value": "flex-end",
              },
              Object {
                "label": "center",
                "textColor": undefined,
                "value": "center",
              },
              Object {
                "label": "stretch",
                "textColor": undefined,
                "value": "stretch",
              },
              Object {
                "label": "baseline",
                "textColor": undefined,
                "value": "baseline",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
